io/src/lib.rs

```diff
@@ -202,3 +277,35 @@ impl std::io::Write for Sink {
 /// Returns a sink to which all writes succeed. See [`std::io::sink`] for more info.
 #[inline]
 pub fn sink() -> Sink { Sink }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn buf_read_fill_and_consume_slice() {
+        let data = [0_u8, 1, 3];
+        // Why does this not use deref coercion to get a slice and find the
+        // implementation of std::io::BufRead for slice?
+        let fill = data.fill_buff();
```

Comment:
I don't know whats going on here, I can't get this to find the implementation of `BufRead`?
```bash
error[E0599]: no method named `fill_buff` found for array `[u8; 3]` in the current scope
   --> io/src/lib.rs:288:25
    |
288 |         let fill = data.fill_buff();
    |                         ^^^^^^^^^ method not found in `[u8; 3]`
```

And explicitly using a reference to the 
```rust
        fn slice(data: &[u8]) {
            let fill = data.fill_buff();
        }
        let data = &[0_u8, 1, 3];
        slice(data);
```
```bash
error[E0599]: no method named `fill_buff` found for reference `&[u8]` in the current scope
   --> io/src/lib.rs:288:25
    |
288 |         let fill = data.fill_buff();
    |                         ^^^^^^^^^ method not found in `&[u8]`
```

Reply:
It's one `f`. `fill_buf` not `fill_buff`. Also the reason first one doesn't work is you need `let mut fill = &[0, 1, 3] as &[_];` so that `.fill_buf` can make `&mut &[u8]` from it.

Reply:
Oh my goodness - face palm. I spent like 30 minutes on that.

Reply:
Yeah, something like that happened to me. It seems like we rely on rustc so much that in those few occasions when it doesn't suggest a typo was made we don't consider it.